frontend/catalyst/pennylane_extensions.py

```diff
@@ -113,10 +113,15 @@ class Function:
 
     Args:
         fn (Callable): the function boundary.
+
+    Raises:
+        AssertionError: Invalid function type.
     """
 
     def __init__(self, fn):
-        assert isinstance(fn, Grad), "Function boundaries only supported for gradients."
+        assert isinstance(
+            fn, Grad
+        ), "Function boundaries only supported for gradients."  # What does boundaries mean?
```

Comment:
I think this is reasonable to leave as an assertion as it is just verifying something that our implementation is not correct, not that the user is using our package correctly 👍